test/unit/network/ConnectionRegistryTest.js

```diff
@@ -0,0 +1,139 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const { expect } = require('chai');
+const sinon = require('sinon');
+const sandbox = sinon.createSandbox();
+
+const { ConnectionRegistryImpl } = require('../../../lib/network/ConnectionManager');
+const { Connection } = require('../../../lib/network/Connection');
+const { ConnectionStrategyConfigImpl, ReconnectMode } = require('../../../lib/config/ConnectionStrategyConfig');
+const { RoundRobinLB } = require('../../../lib/util/RoundRobinLB');
+const { UuidUtil } = require('../../../lib/util/UuidUtil');
+const { ClientOfflineError, IOError } = require('../../../lib/core/HazelcastError');
+
+describe('ConnectionRegistryTest', function () {
+
+    const connectionState = {
+        INITIAL: 0,
+        CONNECTED_TO_CLUSTER: 1,
+        INITIALIZED_ON_CLUSTER: 2
+    };
+
+    afterEach(function () {
+        sandbox.restore();
+    });
+
+    describe('getRandomConnection', function () {
+        it('should use load balancer in smart mode', function () {
+            const loadBalancerStub = sandbox.stub(RoundRobinLB.prototype);
+            const connectionRegistry = new ConnectionRegistryImpl(
+                new ConnectionStrategyConfigImpl(),
+                true,
+                loadBalancerStub
+            );
+            connectionRegistry.getRandomConnection();
+            expect(loadBalancerStub.next.called).to.be.true;
+        });
+
+        it('should return first active connection in non-smart mode without using load balancer',
+        function () {
+                const loadBalancerStub = sandbox.stub(RoundRobinLB.prototype);
+                const connectionRegistry = new ConnectionRegistryImpl(
+                    new ConnectionStrategyConfigImpl(),
+                    false,
+                    loadBalancerStub
+                );
+
+                const firstConnection = new sinon.createStubInstance(Connection);
+                connectionRegistry.setConnection(UuidUtil.generate(), firstConnection);
+                connectionRegistry.setConnection(UuidUtil.generate(), new sinon.createStubInstance(Connection));
+                connectionRegistry.setConnection(UuidUtil.generate(), new sinon.createStubInstance(Connection));
+
+                const randomConnection = connectionRegistry.getRandomConnection();
+
+                expect(randomConnection).to.be.equal(firstConnection);
+                expect(loadBalancerStub.next.called).to.be.false;
+            }
+        );
+    });
+
+    describe('checkIfInvocationAllowed', function () {
+        it('returns null when ConnectionState=INITIALIZED_ON_CLUSTER and there are some active connections',
```

Comment:
nit: the convention is to start test name with `should` and keep it as brief as possible (without loosing meaningful context); the same consideration applies to the tests below
```suggestion
        it('should return null when ConnectionState=INITIALIZED_ON_CLUSTER and there are active connections',
```